src/cli/proxy_mode.rs

```diff
@@ -38,12 +38,22 @@ pub fn main() -> Result<()> {
         } else {
             env::args_os().skip(2).collect()
         };
+        let help = cmd_args.is_empty()
+            || cmd_args == &["help"]
```

Comment:
Do you want this to happen if someone does `cargo help build` ?  Maybe, maybe not.

Reply:
I couldn't have worded it better...

But now I think it shouldn't even hook onto `cargo help`. Once you dive into subcommands, you're on your own. And of course it avoids the awkward output of `rustc help`.

Reply:
And by the way, there's no reason to expect extra flags besides -h or --help. `cargo -V -h` does not give you both version and help, and `cargo --color=always -h` does not colour your day. Help overrules almost everything, including  `cargo -h --baloney`, but not `cargo --baloney -h`.

But that still means it's better to compare the first argument and allow more arguments.